app/models/pieces/pawn.rb

```diff
@@ -1,8 +1,36 @@
 class Pawn < Piece
   before_create :set_defaults
+  #attr_accessor :first_move
   
   def set_defaults
     self.icon = (self.color == 1) ? "&#9823;" : "&#9817;"
+    self.first_move ||= true
   end
-
+  
+  def valid_move?(x, y)
+    # Valid parameters passed in?
+    valid_params = super
+    return false if !valid_params
+    
+    x_diff = self.x_position - x
+    y_diff = self.y_position - y
+    
+    if x_diff == 2 and y_diff == 0
+      return true if self.first_move and !self.game.piece_on?(x,y) and !self.is_obstructed?(x,y)
```

Comment:
try to put the whole if statement into a separated method and give it a name which indicates the case it is checking.. so the return statement would be:

return true if my_new_method_with_a_descriptive_name(x,y)